### ISSUE TITLE
Reorganize directories

### ISSUE THREAD

**Issue by RadixSeven** [NCBI-Hackathons/Network_SNPs#6]
Directory structure is crazy. We need to agree on a structure and move things around. 

```
find ~/ubuntu -type f -print0 | xargs -0 grep -l old_dir_name
```

Should give an idea of which files have dependencies on the old structure.
(note that the command above hasn't been actually run and may contain typos)

**Comment by graphanalytics** [comment]
I totally agree.
Would  you mind proposing a structure that is optimal in your opinion ?

**Comment by RadixSeven** [comment]
I can't propose optional, but I think an improvement would be:

```
~/analyzer/
~/analyzer/hotnet2
~/analyzer/funseq
~/analyzer/better_name_for_networkx
~/analyzer/dmGWAS
~/analyzer/..........
~/meta-analysis -- where converters and meta-net-var.py live
~/networks
~/sample_data
~/docs
```

Is this missing anything you can see?

**Comment by graphanalytics** [comment]
This is great. I would not change a thing in your proposed structure. 
could we rename funseq to funseq2 ?
I propose - simplenetvar - as a better name for oscar's module - as it basically does the simple/straightforward/generic network analysis of variants. 
Wonder how this structure could be represented in the github page.
Team, please do exchange  your thoughts on this.

**Comment by DCGenomics** [comment]
SGTM!

On Thu, Jan 7, 2016 at 12:08 PM, graphanalytics notifications@github.com
wrote:

> This is great. I would not change a thing in your proposed structure.
> could we rename funseq to funseq2 ?
> I propose - simplenetvar - as a better name for oscar's module - as it
> basically does the simple/straightforward/generic network analysis of
> variants.
> Wonder how this structure could be represented in the github page.
> Team, please do exchange your thoughts on this.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NCBI-Hackathons/Network_SNPs/issues/6#issuecomment-169732305
> .

## 

What have you done today to make the world a better place?

**Comment by RadixSeven** [comment]
As a tool for representing the directory structure, I just ran across:

http://www.gravizo.com/

which is tailor-made for our purpose:

![Directory structure](http://g.gravizo.com/g?
 digraph G {
   home -> analyzer;
   home -> meta_analysis;
   meta_analysis [label="meta-analysis"];
   home -> networks;
   home -> sample_data;
   home -> docs;
   analyzer -> hotnet2;
   analyzer -> funseq2;
   analyzer -> simplenetvar;
   analyzer -> dmgwas;
   analyzer -> more;
 })

The source for the above is:

```
![Directory structure](http://g.gravizo.com/g?
 digraph G {
   home -> analyzer;
   home -> meta_analysis;
   meta_analysis [label="meta-analysis"];
   home -> networks;
   home -> sample_data;
   home -> docs;
   analyzer -> hotnet2;
   analyzer -> funseq2;
   analyzer -> simplenetvar;
   analyzer -> dmgwas;
   analyzer -> more;
 }
)
```

**Comment by graphanalytics** [comment]
Looks cool. 
Thanks for sharing this.

**Comment by hagenaue** [comment]
Beautiful!  Should we create a data input and output folder too?

On Thu, Jan 7, 2016 at 12:44 PM, graphanalytics notifications@github.com
wrote:

> Looks cool.
> Thanks for sharing this.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/NCBI-Hackathons/Network_SNPs/issues/6#issuecomment-169752284
> .

## 

Megan Hastings Hagenauer, Ph.D.

Research Investigator in the Molecular, Behavioral Neuroscience Institute
Instructor in the Department of Psychology
University of Michigan-Ann Arbor

**Comment by RadixSeven** [comment]
@hagenaue I don't think so. Besides test inputs, we are not providing any inputs. The user should feel free to put their inputs and outputs where they like. If I were the user, I'd probably structure things by analysis like this:

![Directory structure](http://g.gravizo.com/g?
 digraph G {
   home -> cure_cancer;
   home -> cure_cystic_fibrosis;
   cure_cancer->ca_networks;
   cure_cancer->ca_snps;
   cure_cancer->ca_outputs;
   cure_cystic_fibrosis->cf_networks;
   cure_cystic_fibrosis->cf_snps;
   cure_cystic_fibrosis->cf_outputs;
   cf_networks [label=networks];
   cf_snps [label=snps];
   cf_outputs [label=outputs];
   ca_networks [label=networks];
   ca_snps [label=snps];
   ca_outputs [label=outputs];
 })

That way, all of my data for my two nobel-prize-worthy publications would all be in one place.

However other users might prefer to structure it by file-type as you suggested:
![Directory structure](http://g.gravizo.com/g?
 digraph G {
   home -> networks -> n_cure_cancer; 
   home -> snps -> s_cure_cancer;
   home -> outputs -> o_cure_cancer;
   networks -> n_cure_cystic_fibrosis;
   snps -> s_cure_cystic_fibrosis;
   outputs -> o_cure_cystic_fibrosis;
   n_cure_cancer [label=cure_cancer]
   s_cure_cancer [label=cure_cancer]
   o_cure_cancer [label=cure_cancer]
   n_cure_cystic_fibrosis [label=cure_cystic_fibrosis]
   s_cure_cystic_fibrosis [label=cure_cystic_fibrosis]
   o_cure_cystic_fibrosis [label=cure_cystic_fibrosis]
 })

I think that each of these ways are equally valid and we should let the user choose rather than encoding one of them directly in the repository.

**Comment by graphanalytics** [comment]
should the root of the structure be /home/ubuntu/ rather than /home/ ?
because /home/ would be the system level directory in unix...
i created analyzer withing /home/ubuntu/ ..... does that make sense ?

**Comment by nyetsche** [comment]
Yes, it should be `/home/ubuntu`, but actually, I'd suggest even rewriting it to be `$HOME/Network_SNPs`, (or a pure relative directory), so the Github checkout works even when not using AWS.

**Comment by hagenaue** [comment]
/home/ubuntu didn't seem to work, so I have left it with the default, which is outputting to the vegas folder.  Since it is only output from a sub-step in the pipeline (not final output), I figure it is probably not worth picking at anymore.
